pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -142,6 +144,112 @@ def _save_input_params(self, idata) -> None:
+        return prior_distribution
+
+    def _create_likelihood_distribution(
+        self,
+        dist: Dict,
+        mu: SharedVariable,
+        observed: Union[np.ndarray, pd.Series],
+        dims: str,
+    ) -> TensorVariable:
+        """
+        Create and return a likelihood distribution for the model.
+
+        This method prepares the distribution and its parameters as specified in the
+        configuration dictionary, validates them, and constructs the likelihood
+        distribution using PyMC.
+
+        Parameters
+        ----------
+        dist : Dict
+            A configuration dictionary that must contain a 'dist' key with the name of
+            the distribution and a 'kwargs' key with parameters for the distribution.
+        observed : Union[np.ndarray, pd.Series]
+            The observed data to which the likelihood distribution will be fitted.
+        dims : str
+            The dimensions of the data.
+
+        Returns
+        -------
+        TensorVariable
+            The likelihood distribution constructed with PyMC.
+
+        Raises
+        ------
+        ValueError
+            If 'kwargs' key is missing in `dist`, or the parameter configuration does
+            not contain 'dist' and 'kwargs' keys, or if 'mu' is present in the nested
+            'kwargs'
+        """
+        # Validate that 'kwargs' is present and is a dictionary
+        if "kwargs" not in dist or not isinstance(dist["kwargs"], dict):
+            raise ValueError(
+                "The 'kwargs' key must be present in the 'dist' dictionary and be a dictionary itself."
+            )
+
+        # Validate that each value in 'kwargs' is a dictionary with the required keys
+        for param, param_config in dist["kwargs"].items():
+            if (
+                not isinstance(param_config, dict)
+                or "dist" not in param_config
+                or "kwargs" not in param_config
+            ):
+                raise ValueError(
+                    f"The parameter configuration for '{param}' must be a dictionary containing 'dist' and 'kwargs' keys."
+                )
+
+            if "mu" in param_config["kwargs"]:
+                raise ValueError(
+                    f"The 'mu' key is not allowed within 'kwargs' of parameter '{param}' as it is reserved."
+                )
+
+        # Extract the likelihood distribution name and instantiate it without passing it as a positional argument
+        likelihood_dist_name = dist["dist"]
+        likelihood_dist = self._get_distribution(dist={"dist": likelihood_dist_name})
+
+        # Prepare parameter distributions directly from 'kwargs'
+        parameter_distributions = {
+            param: self._get_distribution(dist=param_config)(
+                **param_config["kwargs"], name=f"likelihood_{param}"
+            )
+            for param, param_config in dist["kwargs"].items()
+        }
```

Comment:
Won't this fail if someone tries to do a `StudentT` likelihood with fixed `nu=4`? Seems like the logic is assuming any other kwargs will always be a `dist`

Reply:
I attempted to understand the model by allowing users to add values such as "nu=3" or "sigma=5". However, after several attempts, I noticed that this approach does not result in the model converging well and sometimes leads to divergences. Although I am open to the possibility, I am a little skeptical based on my results. As long as the priors are defined by distributions, I have never encountered any problems.

Reply:
Modelling wise it's very hard to infer the nu of a StudentT and it's usually advised to choose a constant or be rather wise about the prior. 

In the general case we should allow constants. It's  too restrictive to force everything to be a parameter.

Reply:
Perfect, makes total sense! I'll work on add the changes